### ISSUE TITLE
pyam.IamDataFrame.to_excel not working

### ISSUE THREAD

**Issue by Chilipp** [IAMconsortium/pyam#149]
Inline with review https://github.com/openjournals/joss-reviews/issues/1095 I went through the [first steps notebook](http://software.ene.iiasa.ac.at/pyam/tutorials/pyam_first_steps.html) and realized that the to_excel method does not export the data when calling `df.to_excel('tutorial_export.xlsx')` at the end.

I installed pyam via ``conda create -n pyam -c conda-forge pyam`` and the documentation requirements via ``pip install -r doc/requirements.txt``

Can you confirm this behaviour? I attached below some system information and the conda environment packages.

<details>
<summary>conda info</summary>

```
     active environment : pyam
    active env location : /Users/psommer/miniconda3/envs/pyam
            shell level : 2
       user config file : /Users/psommer/.condarc
 populated config files : /Users/psommer/.condarc
          conda version : 4.5.11
    conda-build version : 3.11.0
         python version : 3.6.4.final.0
       base environment : /Users/psommer/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /Users/psommer/miniconda3/pkgs
                          /Users/psommer/.conda/pkgs
       envs directories : /Users/psommer/miniconda3/envs
                          /Users/psommer/.conda/envs
               platform : osx-64
             user-agent : conda/4.5.11 requests/2.18.4 CPython/3.6.4 Darwin/17.7.0 OSX/10.13.6
                UID:GID : 502:20
             netrc file : None
           offline mode : False
```
</details>

<details>
<summary>conda list -n pyam</summary>

```
# packages in environment at /Users/psommer/miniconda3/envs/pyam:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                    <pip>
appnope                   0.1.0                     <pip>
asn1crypto                0.24.0                py36_1003    conda-forge
attrs                     18.2.0                     py_0    conda-forge
Babel                     2.6.0                     <pip>
backcall                  0.1.0                     <pip>
blas                      1.0                         mkl  
bleach                    3.0.2                     <pip>
boost-cpp                 1.68.0               h3a22d5f_0    conda-forge
bzip2                     1.0.6                         1    conda-forge
ca-certificates           2018.11.29           ha4d7672_0    conda-forge
cairo                     1.14.12              h276e583_5    conda-forge
cartopy                   0.17.0           py36h81b52dc_0    conda-forge
certifi                   2018.11.29            py36_1000    conda-forge
cffi                      1.11.5           py36h5e8e0c9_1    conda-forge
chardet                   3.0.4                 py36_1003    conda-forge
click                     7.0                        py_0    conda-forge
click-plugins             1.0.4                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
cloud-sptheme             1.9.4                     <pip>
cryptography              2.3.1            py36hdffb7b8_0    conda-forge
cryptography-vectors      2.3.1                 py36_1000    conda-forge
curl                      7.62.0               h74213dd_0    conda-forge
cycler                    0.10.0                     py_1    conda-forge
decorator                 4.3.0                     <pip>
descartes                 1.1.0                      py_2    conda-forge
docutils                  0.14                      <pip>
entrypoints               0.2.3                     <pip>
expat                     2.2.5                hfc679d8_2    conda-forge
fiona                     1.8.3            py36hfc77a4a_0    conda-forge
fontconfig                2.13.1               hce039c3_0    conda-forge
freetype                  2.9.1                h6debe1e_4    conda-forge
freexl                    1.0.5                h470a237_2    conda-forge
gdal                      2.3.2            py36hb00a9d7_0    conda-forge
geopandas                 0.4.0                      py_1    conda-forge
geos                      3.6.2                hfc679d8_4    conda-forge
geotiff                   1.4.2                h700e5ad_5    conda-forge
gettext                   0.19.8.1             h1f1d5ed_1    conda-forge
giflib                    5.1.4                h470a237_1    conda-forge
glib                      2.56.2               h464dc38_1    conda-forge
hdf4                      4.2.13               h951d187_2    conda-forge
hdf5                      1.10.3               hc401514_2    conda-forge
icu                       58.2                 hfc679d8_0    conda-forge
idna                      2.7                   py36_1002    conda-forge
imagesize                 1.1.0                     <pip>
intel-openmp              2019.1                      144  
ipykernel                 5.1.0                     <pip>
ipython                   7.2.0                     <pip>
ipython-genutils          0.2.0                     <pip>
ipywidgets                7.4.2                     <pip>
jedi                      0.13.1                    <pip>
Jinja2                    2.10                      <pip>
jpeg                      9c                   h470a237_1    conda-forge
json-c                    0.12.1               h470a237_1    conda-forge
jsonschema                2.6.0                     <pip>
jupyter                   1.0.0                     <pip>
jupyter-client            5.2.3                     <pip>
jupyter-console           6.0.0                     <pip>
jupyter-core              4.4.0                     <pip>
kealib                    1.4.10               hb88cf67_0    conda-forge
kiwisolver                1.0.1            py36h2d50403_2    conda-forge
krb5                      1.16.2               hbb41f41_0    conda-forge
latexcodec                1.0.5                     <pip>
libcurl                   7.62.0               hbdb9355_0    conda-forge
libcxx                    7.0.0                h2d50403_2    conda-forge
libdap4                   3.19.1               h18059cb_1    conda-forge
libedit                   3.1.20170329         haf1bffa_1    conda-forge
libffi                    3.2.1                hfc679d8_5    conda-forge
libgdal                   2.3.2                h6a28ee2_0    conda-forge
libgfortran               3.0.1                h93005f0_2  
libiconv                  1.15                 h470a237_3    conda-forge
libkml                    1.3.0                he469717_9    conda-forge
libnetcdf                 4.6.1               h350cafa_11    conda-forge
libpng                    1.6.36               ha92aebf_0    conda-forge
libpq                     10.5                 hf16a0db_1    conda-forge
libspatialindex           1.8.5                hfc679d8_3    conda-forge
libspatialite             4.3.0a              h3b29d86_23    conda-forge
libssh2                   1.8.0                h5b517e9_3    conda-forge
libtiff                   4.0.10               he6b73bb_0    conda-forge
libxml2                   2.9.8                h422b904_5    conda-forge
libxslt                   1.1.32               h88dbc4e_2    conda-forge
llvm-meta                 7.0.0                         0    conda-forge
lxml                      4.2.5            py36hc9114bc_0    conda-forge
MarkupSafe                1.1.0                     <pip>
matplotlib                3.0.2                    py36_1    conda-forge
matplotlib-base           3.0.2            py36hb2d221d_1    conda-forge
mistune                   0.8.4                     <pip>
mkl                       2018.0.3                      1  
mkl_fft                   1.0.10                   py36_0    conda-forge
mkl_random                1.0.2                    py36_0    conda-forge
munch                     2.3.2                      py_0    conda-forge
nbconvert                 5.3.1                     <pip>
nbformat                  4.4.0                     <pip>
nbsphinx                  0.3.5                     <pip>
ncurses                   6.1                  hfc679d8_1    conda-forge
notebook                  5.7.2                     <pip>
numpy                     1.15.4           py36h6a91979_0  
numpy-base                1.15.4           py36h8a80b8c_0  
numpydoc                  0.8.0                     <pip>
olefile                   0.46                       py_0    conda-forge
openjpeg                  2.3.0                h316dc23_3    conda-forge
openssl                   1.0.2p               h470a237_1    conda-forge
oset                      0.1.3                     <pip>
owslib                    0.17.0                     py_0    conda-forge
packaging                 18.0                      <pip>
pandas                    0.23.4           py36hf8a1672_0    conda-forge
pandocfilters             1.4.2                     <pip>
parso                     0.3.1                     <pip>
patsy                     0.5.1                      py_0    conda-forge
pcre                      8.41                 hfc679d8_3    conda-forge
pexpect                   4.6.0                     <pip>
pickleshare               0.7.5                     <pip>
pillow                    5.3.0            py36hc736899_0    conda-forge
pip                       18.1                  py36_1000    conda-forge
pixman                    0.34.0               h470a237_3    conda-forge
poppler                   0.67.0               hdf8a1b3_2    conda-forge
poppler-data              0.4.9                         0    conda-forge
postgresql                10.5                 ha408888_1    conda-forge
proj4                     4.9.3                h470a237_8    conda-forge
prometheus-client         0.4.2                     <pip>
prompt-toolkit            2.0.7                     <pip>
psycopg2                  2.7.6.1          py36hdffb7b8_0    conda-forge
ptyprocess                0.6.0                     <pip>
pyam                      0.1.2                         0    conda-forge
pybtex                    0.22.0                    <pip>
pybtex-docutils           0.2.1                     <pip>
pycparser                 2.19                       py_0    conda-forge
pyepsg                    0.4.0                      py_0    conda-forge
Pygments                  2.3.0                     <pip>
pykdtree                  1.3.1            py36h7eb728f_2    conda-forge
pyopenssl                 18.0.0                py36_1000    conda-forge
pyparsing                 2.3.0                      py_0    conda-forge
pyproj                    1.9.5.1          py36h508ed2a_6    conda-forge
pysal                     1.14.4.post2          py36_1001    conda-forge
pyshp                     2.0.0                      py_0    conda-forge
pysocks                   1.6.8                 py36_1002    conda-forge
python                    3.6.7                h5001a0f_1    conda-forge
python-dateutil           2.7.5                      py_0    conda-forge
pytz                      2018.7                     py_0    conda-forge
pyyaml                    3.13             py36h470a237_1    conda-forge
pyzmq                     17.1.2                    <pip>
qtconsole                 4.4.3                     <pip>
readline                  7.0                  haf1bffa_1    conda-forge
requests                  2.20.1                py36_1000    conda-forge
rtree                     0.8.3                 py36_1000    conda-forge
scipy                     1.1.0            py36h28f7352_1  
seaborn                   0.9.0                      py_0    conda-forge
Send2Trash                1.5.0                     <pip>
setuptools                40.6.2                   py36_0    conda-forge
shapely                   1.6.4            py36h164cb2d_1    conda-forge
six                       1.11.0                py36_1001    conda-forge
snowballstemmer           1.2.1                     <pip>
Sphinx                    1.8.2                     <pip>
sphinx-gallery            0.2.0                     <pip>
sphinxcontrib-bibtex      0.4.1                     <pip>
sphinxcontrib-fulltoc     1.2.0                     <pip>
sphinxcontrib-programoutput 0.11                      <pip>
sphinxcontrib-websupport  1.1.0                     <pip>
sqlalchemy                1.2.14           py36h470a237_0    conda-forge
sqlite                    3.26.0               hb1c47c0_0    conda-forge
statsmodels               0.9.0            py36h7eb728f_0    conda-forge
terminado                 0.8.1                     <pip>
testpath                  0.4.2                     <pip>
tk                        8.6.9                ha92aebf_0    conda-forge
tornado                   5.1.1            py36h470a237_0    conda-forge
traitlets                 4.3.2                     <pip>
urllib3                   1.23                  py36_1001    conda-forge
wcwidth                   0.1.7                     <pip>
webencodings              0.5.1                     <pip>
wheel                     0.32.3                   py36_0    conda-forge
widgetsnbextension        3.4.2                     <pip>
xerces-c                  3.2.0                h5d6a6da_2    conda-forge
xlrd                      1.1.0                      py_2    conda-forge
xlsxwriter                1.1.2                      py_0    conda-forge
xz                        5.2.4                h470a237_1    conda-forge
yaml                      0.1.7                h470a237_1    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge
```
</details>

**Comment by gidden** [comment]
Hey @Chilipp, thanks for raising this issue. I just tested on my linux machine, and all seems fine. Our CI tests on all three OSes, so I opened #151 to start to investigate.

**Comment by gidden** [comment]
closed by #151

**Comment by Chilipp** [comment]
All right, thanks. It worked for me as well.
